etc/ha_healthchecker/ha_healthchecker.py.j2

```diff
@@ -157,35 +160,27 @@ class Refresher(Base):
                                    **update_dict)
 
     def _need_fix_alarmed_status(self, node):
-        service_objs = self.zk.list_services(node_name_filter=node.name)
-        err_svcs = []
-        for service_obj in service_objs:
-            if service_obj.status == 'down':
-                err_svcs.append(service_obj)
+        if not node.alarmed:
+            return False
 
-        need_fix = False
-        if len(err_svcs) == 0:
-            need_fix = True
-        else:
-            fix_res = []
-            for err_svc in err_svcs:
-                if (not err_svc.is_necessary and
-                        not self._is_alarmed_timeout(err_svc) and
-                        node.role == 'master'):
-                    fix_res.append(True)
-            if len(err_svcs) == len(fix_res):
-                need_fix = True
+        if node.role == 'slave':
+            return True
+
+        for service in self.zk.list_services(node_name_filter=node.name):
+            if service.is_necessary:
+                continue
+            if self._is_alarmed_timeout(service):
```

Comment:
we can not only see the alarmed_at, a service can be alarmed=False and alarmed_at !=None

Reply:
Done